service/src/main/java/com/codeforcommunity/dataaccess/AuthDatabaseOperations.java

```diff
@@ -90,4 +93,47 @@ public boolean isOnBlackList(String signature) {
             Tables.BLACKLISTED_REFRESHES
                 .where(Tables.BLACKLISTED_REFRESHES.REFRESH_HASH.eq(signature)));
     }
+
+    public void createSecretKey(int userId, String token) throws AuthException {
+        if (!db.fetchExists(Tables.NOTE_USER.where(NOTE_USER.ID.eq(userId)))) {
+            throw new AuthException("User does not exist.");
+        }
+
+        VerificationKeysRecord keysRecord = db.newRecord(Tables.VERIFICATION_KEYS);
+        keysRecord.setId(token);
+        keysRecord.setUserId(userId);
+        keysRecord.store();
+    }
+
+    private boolean isTokenDateValid(VerificationKeysRecord tokenResult) {
+        Timestamp cutoffDate = Timestamp.from(Instant.now().minusSeconds(CUTOFF_TIME));
+        if (tokenResult.getCreated().before(cutoffDate)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public void validateSecretKey(String secretKey) throws AuthException {
+        VerificationKeysRecord veriKey = db.selectFrom(Tables.VERIFICATION_KEYS)
+            .where(Tables.VERIFICATION_KEYS.ID.eq(secretKey)
+                .and(Tables.VERIFICATION_KEYS.USED.eq((short)0)))
```

Comment:
There is almost definitely a better way of writing this query than casting 0 to a short

Reply:
jOOQ will create boolean fields for us, you just have to find the right data type to use in the SQL file...